reference-adapter/src/main/java/org/egov/ifix/consumer/EventTypeConsumer.java

```diff
@@ -103,16 +105,25 @@ private void processFiscalEvent(String record) {
             log.error(LOG_ERROR_PREFIX + NON_RECOVERABLE_ERROR +
                     "Exception while transforming and collecting fiscal data. ", e);
 
-            EventPostingDetail errorDetail = new EventPostingDetail();
-            errorDetail.setEventId(eventJsonObject.get(ID).getAsString());
-            errorDetail.setTenantId(eventJsonObject.get(TENANT_ID).getAsString());
-            errorDetail.setEventType(eventJsonObject.get(EVENT_TYPE).getAsString());
-            errorDetail.setCreatedDate(new Date());
-            errorDetail.setLastModifiedDate(new Date());
-            errorDetail.setStatus(NA);
-            errorDetail.setError("Internal Fiscal Event Conversion/Transformation error : " + e.getMessage());
-            errorDetail.setRecord(record);
-            eventPostingDetailRepository.save(errorDetail);
+            EventMapper eventMapper = eventTypeMap.get(eventJsonObject.get(EVENT_TYPE).getAsString());
+            List<String> referenceIdList = eventMapper.getReferenceIdList(jsonObject);
+            List<EventPostingDetail> eventPostingDetailList = new ArrayList<>();
+
+            for (String referenceId : referenceIdList) {
+                EventPostingDetail errorDetail = new EventPostingDetail();
```

Comment:
Instead of creating a new empty object, could you create it from a template and just fill up the one field that is different - referenceId.